### ISSUE TITLE
E3035 Strict checking for DeletionPolicy that includes JINJA2

### ISSUE THREAD

**Issue by jitesh88** [aws-cloudformation/cfn-python-lint#793]
*cfn-lint version: 0.14

*Description of issue.*
There is a bug with the DeletionPolicy for RDS where you cannot pass in parameter reference or do any substitutions. It expects hardcoded static values i.e. Delete, Retain

https://forums.aws.amazon.com/thread.jspa?start=25&threadID=152800&tstart=0

This doesn't serve us well when trying to deploy the same template across different environments i.e. dev/stg/prd

to address this problem, we use Jinja2 Templating as below:
```yaml
  RDSDBInstance:
    Type: AWS::RDS::DBInstance
    DeletionPolicy: "{{ Deletion_Policy }}"
    Properties:
```
Since the release of version 0.14 E3035 rule was added to check for deletion policy values.

it fails to validate our template with the below error:

`E3035 DeletionPolicy should be only one of Delete, Retain, Snapshot at Resources/RDSDBInstance/DeletionPolicy`

**Comment by cmmeyer** [comment]
cfn-lint isn't aware of JINJA2, mustache or other template wrappers beyond CloudFormation JSON/YAML.

I'm going to flag this as an enhancement in case we ever decide to incorporate JINJA2 pre-processing, but for now you'll want to run your templates through the JINJA2 pre-processor before validating with cfn--lint.

**Comment by PatMyron** [comment]
Closing this with Chuck's recommendation as it won't be feasible for the Linter to handle all pre-processing intentions

[tracking here for official CloudFormation support for non-literal `DeletionPolicy`/`UpdateReplacePolicy`](https://github.com/aws-cloudformation/aws-cloudformation-coverage-roadmap/issues/162)
